fastlane_core/lib/fastlane_core/feature/feature_manager.rb

```diff
@@ -0,0 +1,53 @@
+module FastlaneCore
+  class FeatureManager
+    class << self
+      attr_accessor :enabled_features
+    end
+    @enabled_features = []
+
+    def self.experiments_enabled?
+      return ENV['FASTLANE_ENABLE_ALL_EXPERIMENTS']
+    end
+
+    def self.enabled?(key)
+      feature = features.detect { |feat| feat.key == key }
+      return false if feature.nil?
+      return true if experiments_enabled? && feature.experiment == true
+      return @enabled_features.include?(key) || ENV[feature.env_var]
+    end
+
+    def self.register_class_method(klass: nil, symbol: nil, default_symbol: nil, override_symbol: nil, key: nil)
+      return if klass.nil? || symbol.nil? || default_symbol.nil? || override_symbol.nil? || key.nil?
+      klass.define_singleton_method(symbol) do |*args|
+        if enabled?(key)
+          klass.send(override_symbol, *args)
+        else
+          klass.send(default_symbol, *args)
```

Comment:
Rather than framing this in terms of "default_symbol" and "override_symbol", I think it may be easier to follow if they're phrased as "enabled_symbol" and "disabled_symbol", or similar... what do you think?